chia/simulator/setup_services.py

```diff
@@ -39,6 +40,16 @@
 log = logging.getLogger(__name__)
 
 
+@contextmanager
+def create_lock_and_load_config(certs_path: Path, root_path: Path) -> Iterator[Dict[str, Any]]:
+    init(None, root_path)
+    init(certs_path, root_path)
+    path = config_path_for_filename(root_path=root_path, filename="config.yaml")
+    path.write_text(path.read_text().replace("localhost", "127.0.0.1"))
```

Comment:
it doesn't look like this only affects tests. does this make sense to do in production too?

Reply:
Actually this is meant to be for tests only, there was some flakiness issue a while ago with using `localhost` and then we switched over to using the IP instead, see https://github.com/chia-network/chia-blockchain/pull/13324.

In this case here it's not about flakiness but rather about the recent PR to resolve the peer outside the reconnect task which we don't do in tests hence the two PRs linked in here fail in CI without this PR.

I guess in production we don't want to get rid of the `localhost` because we also want to cover ipv6?

Reply:
ok, I didn't realize this was in the simulator at first. So, this primarily affects tests. However, I would expect it to also affect users of the "stand-alone" simulator. It probably doesn't matter much though